resources/pre-upgrade.sh

```diff
@@ -3,6 +3,14 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-# TODO: Only dump database if necessary
-# e.g. at major upgrades, not at bugfix upgrades
-pg_dumpall -U postgres -f "${PGDATA}"/postgresqlFullBackup.dump
+FROM_VERSION="${1}"
+TO_VERSION="${2}"
+FROM_MAJOR_VERSION=$(echo "${FROM_VERSION}" | cut -d '.' -f1)
+TO_MAJOR_VERSION=$(echo "${TO_VERSION}" | cut -d '.' -f1)
+
+# dump database if TO_MAJOR_VERSION is equal or higher than 12 and FROM_MAJOR_VERSION is lower than 12
+if [ "12" == "$(printf "%s\\n12" "${TO_MAJOR_VERSION}" | sort -n | head -n1)" ] && [ "12" != "$(printf "%s\\n12" "${FROM_MAJOR_VERSION}" | sort -n | head -n1)" ]; then
```

Comment:
I have trouble to understand this line, even with the comment above.

The comment line itself is a giveaway that this whole test should be extracted into a function.
You already split off the major version, so basically you work on integers here. Did you consider the bash tests `-lt / -le 
/ -eq / -ge / -gt`? I can imagine that this would make the actual test more readable, something along the lines of:

```bash
if [[ ${TO_MAJOR_VERSION} -eq 12 ]] && isMajorVersionUpgrade ${FROM_MAJOR_VERSION} ${TO_MAJOR_VERSION} ; then
  # dump all the things
fi

function isMajorVersionUpgrade() {
  local fromMajorVersion=$1
  local toMajorVersion=$2

  if [[ ${fromMajorVersion} -lt ${toMajorVersion} ]];
    return 0
  fi
  return 1
}
```

- [ ] done

Reply:
Totally right, this could be much simpler. I changed it.